client/blocks/ai-translations-banner/index.jsx

```diff
@@ -0,0 +1,55 @@
+import { getLanguage, isAITranslatedLocale, isDefaultLocale } from '@automattic/i18n-utils';
+import { useTranslate } from 'i18n-calypso';
+import Banner from 'calypso/components/banner';
+import QueryLocaleSuggestions from 'calypso/components/data/query-locale-suggestions';
+import { useSelector } from 'calypso/state';
+import getLocaleSuggestions from 'calypso/state/selectors/get-locale-suggestions';
+
+const BANNER_NAME = 'ai_translations_banner';
+
+export default function AITranslationsBanner() {
+	const translate = useTranslate();
+	const localeSuggestions = useSelector( getLocaleSuggestions );
+	const isCurrentlyUsingAITranslatedLocale = isAITranslatedLocale( translate.localeSlug );
+
+	if ( ! isDefaultLocale( translate.localeSlug ) && ! isCurrentlyUsingAITranslatedLocale ) {
+		return null;
+	}
+
+	if ( ! localeSuggestions ) {
+		return <QueryLocaleSuggestions />;
+	}
+
+	if ( ! isAITranslatedLocale( localeSuggestions?.[ 0 ]?.locale ) ) {
+		return;
+	}
+
+	const language = getLanguage( localeSuggestions?.[ 0 ]?.locale );
```

Comment:
In my test case I've changed my WordPress.com locale to Polish and my browser languages were set to Norwegian, Thai and Polish. Going to /home I got a banner asking to switch to Norwegian without a CTA. I think that it would be more consistent to change the logic to:
1. If user locale is English and localeSuggestions contains AI Translated Locale - show the banner for the first suggested locale with the CTA.
2. If user locale is contained in AI Translated Locale - show the banner for the user locale without the CTA.
<img width="872" alt="Screenshot 2024-12-11 at 00 18 25" src="https://github.com/user-attachments/assets/e2c2cb87-ff90-43ed-afd0-f85d0c573fa8">

Reply:
Thanks for the suggestion, it makes sense! I've updated the PR.

Reply:
Works pretty well! I did find two more edge cases:
1. When the `Display interface in English` checkbox is selected, the notification isn't being shown.
2. If my UI locale is set to an AI Translated Locale, but locale-guess doesn't return any of the AI translated locales, the notification also won't be shown.

Reply:
Here're some test cases I can think of:
1. UI locale is English, browser locales contain one of the AI locales - show the prompt for top detected browser locale, with CTA.
2. UI locale is English, browser locales don't contain any of the AI locales - don't show the prompt
3. UI locale is AI locale without the 'show english' checkbox, browser locales contain the AI locale - show the prompt for user locale, without CTA.
4. UI locale is AI locale without the 'show english' checkbox, browser locales contain other AI locale - show the prompt for user locale, without CTA.
5. UI locale is AI locale without the 'show english' checkbox, browser locales contain no AI locales - show the prompt for user locale, without CTA.
6. UI locale is AI locale with the 'show english' checkbox - same as 3-5?

Reply:
> Could you provide the exact steps you used for 1?

Can't reproduce it right now, so it might've a different case. I do see an issue with test case 7 (user locale is Norwegian with "show english" checkbox, browser locales are Thai, German, Dutch) where the notification shows a prompt for Thai with a CTA.

> Updated the logic to handle the case.

Test case 5 fails for me. Steps to reproduce: sandbox public-api with code-D167396, set user locale to Norwegian Bokmal, set Chrome top browser languages to German / Dutch / Japanese, open the site home for a simple site. locale-guess API call returns de/ja/nl and the site home shows no notification and some card stuck in loading state. Tested both on calypso.live and locale instance.

<img width="836" alt="Screenshot 2024-12-12 at 21 10 55" src="https://github.com/user-attachments/assets/bad25930-ec30-4c42-9bc5-fe80b2e4bfd7" />

Lastly, are the clicks on support link and /me/account link being captured as track events?

Reply:
> Lastly, are the clicks on support link and /me/account link being captured as track events?

Yes, [CTA clicks are tracked](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/banner/index.jsx#L132-L139) as well as dismissing the notice.

Reply:
> Can't reproduce it right now, so it might've a different case. I do see an issue with test case 7 (user locale is Norwegian with "show english" checkbox, browser locales are Thai, German, Dutch) where the notification shows a prompt for Thai with a CTA.

I switched to reading the locale from the user object, rather than reading it from `i18n-calypso`. With this change, we should now have the expected behavior with the "Display interface in English" checkbox.

> Test case 5 fails for me. Steps to reproduce: sandbox public-api with code-D167396, set user locale to Norwegian Bokmal, set Chrome top browser languages to German / Dutch / Japanese, open the site home for a simple site. locale-guess API call returns de/ja/nl and the site home shows no notification and some card stuck in loading state. Tested both on calypso.live and locale instance.

I followed the steps for test case 5 and it works as expected. The infinite loading state of the card below that you mentioned is unrelated to the translations notice banner, but it makes me think that there might be some other problem on your end. Could you inspect the network request and see if anything is failing there?